### ISSUE TITLE
Look into asynchronous views

### ISSUE THREAD

**Issue by cc-a** [ImperialCollegeLondon/drunc_ui#140]
We are currently using standard syncronous views in Django. This leads to us needing to wrap the asyncronous interface to the DRUNC process manager.

It would be nice to use asynchronous views in order to:
- Reduce the boiler plate of wrapping methods of the `ProcessManagerDriver` class.
- Exploit performance benefits. This especially may be useful as we expect the multiple simultaneous frontends to be polling our server and asyncronous views would help defray the performance implications of interacting with the process manager.

Need to have a careful look at this as we've not done it before and there are several caveats noted in the Django docs - https://docs.djangoproject.com/en/5.1/topics/async/.

**Comment by cc-a** [comment]
Just stumbled across [Daphne](https://docs.djangoproject.com/en/5.1/howto/deployment/asgi/daphne/) which looks like it will be useful for testing async during development.

**Comment by cc-a** [comment]
Just noting that we have at least piece of sync only middleware - whitenoise. There is a whitenoise fork called [ServeStatic](https://archmonger.github.io/ServeStatic/latest/) which has async support and looks promising.

**Comment by cc-a** [comment]
This blog is an interesting read on the topic https://jonathanadly.com/is-async-django-ready-for-prime-time

**Comment by TinyMarsh** [comment]
After exploring this issue and some discussion, we decided that while async views could reduce boilerplate and improve performance, we've decided to put this work on hold for now.

The primary blocker is the non-async nature of Django's template rendering system. Every call to render a template would need to be wrapped in sync_to_async, introducing complexity and potentially negating some of the performance benefits we hoped to gain.

Until Django templates natively support async or we find a way to address this without adding significant overhead, we're shelving this effort. We'll revisit this at a later date, where we could explore alternatives such as third-party async template engines like Jinja2 (which Django supports natively).
